README.md

```diff
@@ -1,4 +1,21 @@
-## Sensor fusion Software based on Principal Component Analysis ##
+# Sensor fusion Software based on Principal Component Analysis #
 
 The purpose of the sensor fusion software is to calculate a single aggregate value from a stream of sensor data input by essentially adapting a sensor fusion algorithm. Sensor fusion algorithm in consideration is “A Simple Multi-Sensor Data Fusion Algorithm Based on Principal Component
 Analysis” The algorithm takes row data from the sensors as inputs, perform Principal Component calculations and returns a value. The value generated by the algorithm is representative of the correct raw data from the sensors (i.e. there may be sensors giving wrong measures). The algorithm is executed every time new sensor data is received. The algorithm receives the data from all the sensors synchronously.
+
+
+## CSV Parser Module ##
+Brief Description of CSV Parser Module: function to read given input csv file and store it into a 2D array and extract all sensor readings for unique time stamps and store it a 2D array with each row representing one unique time stamp
+
+### Known Limitations ###
+This CSV Parser can read maximum of 10 unique time stamps and each time stamp can have maximum of 20 sensor readings, therefore a total of maximum 200 sensor readings
+
+### Usage ### 
+Run the csv_parser.c source file on any C compiler with a sample.csv input file on the same directory as of source file
+
+### Output ###
+
+Full CSV File can be accessed at csv[][3]
+First unique time stamp can be accessed at csv[0][0], Rest of the Individual unique time stamps can be accessed in csv[u_ts[]][0] 
+Sensor readings for each unique time stamps can be accessed at sr[][] at row wise
+Number of sensor readings at each time stamps can be accessed at sr_c[]   

```

Comment:
In your README.md file don't forget to meet the minimum requirements requested by our professor. I've copied them again here for you convenience: 

1.     Use Markdown notation for better readability.
2.     Organization: University name
3.     Authors: 
4.     Brief description of what the program does 
5.     Source File Organization, list of folders and files
6.     Instructions on how to compile/build the code and the test. You should also explain dependencies and refer the user/developer to the "place" where these instructions can be found (if they are too long).
7.     Detailed instructions for the user and developers should be included in the documentation folder. In the readme file, you should specify where to find them

Reply:
Updated according to comments in the dev branch pull request